presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -72,12 +76,15 @@ private static SymbolStatsEstimate addColumnStats(SymbolStatsEstimate leftStats,
         double nullsCountLeft = leftStats.getNullsFraction() * leftRows;
         double totalSizeLeft = (leftRows - nullsCountLeft) * leftStats.getAverageRowSize();
         double totalSizeRight = (rightRows - nullsCountRight) * rightStats.getAverageRowSize();
-        double newNullsFraction = (nullsCountLeft + nullsCountRight) / newRowCount;
+        double newNullsFraction = newRowCount == 0 ? 1.0 : ((nullsCountLeft + nullsCountRight) / newRowCount);
         double newNonNullsRowCount = newRowCount * (1.0 - newNullsFraction);
 
+        // FIXME, weights to average. left and right should be equal in most cases anyway
+        double newAverageRowSize = newNonNullsRowCount == 0 ? 0 : ((totalSizeLeft + totalSizeRight) / newNonNullsRowCount);
```

Comment:
Instead of this we probably should have a special casing in `getOutputSizeForSymbol`.

Actually, we do:
```
        if (isNaN(averageRowSize)) {
            if (type instanceof FixedWidthType) {
                averageRowSize = ((FixedWidthType) type).getFixedSize();
            }
            else {
                averageRowSize = DEFAULT_DATA_SIZE_PER_COLUMN;
            }
        }

        double outputSize = numberOfNonNullRows * averageRowSize;
```
So this change seems not needed

Reply:
Division by zero may result in `INFINITY`. We don't want to set average row size to that.

Reply:
It seems that normalizer could correct such stats (0 rows, `inifity` avg.row size). Then we wouldn't need to add multiple checks, but just a single one in normalizer.

Reply:
Currently infinite row size is not allowed in SymbolStats. This value doesn't make much sense.